### PR TITLE
Solution to week one's weekend challenge.

### PR DESCRIPTION
**Pull request by adc17**

### CODE REVIEW

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+require './lib/plane'
+
+class Airport
+
+  attr_reader :planes, :climate
+  attr_accessor :capacity
+  DEFAULT_CAPACITY = 5
+
+  def initialize(climate = Climate.new)
+    raise 'Error: Parameter is not a climate.' unless climate.is_a?(Climate)
```
Comment (ddosque) [inline lib/airport.rb]: avoid checking type in loosely typed languages - you prevent the evolution of duck types.

File: lib/airport.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+require './lib/climate'
+require './lib/plane'
+
+class Airport
+
+  attr_reader :planes, :climate
+  attr_accessor :capacity
```
Comment (ddosque) [inline lib/airport.rb]: no attr_accessor / reader / writer is required for this implementation. Avoid using these unless they are private.

File: lib/plane.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    confirm_landing
+  end
+
+  def take_off
+    take_off_process
+    confirm_take_off
+  end
+
+  def confirm_landing
+    landed ? "Plane has landed." : "Plane has not landed."
```
Comment (ddosque) [inline lib/plane.rb]: Avoid mixing presentation logic in your 'business' code.

File: spec/airport_spec.rb [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+      it 'raises error' do
+        expect { Airport.new(@not_a_climate) }.to raise_error 'Error: Parameter is not a climate.'
+      end
+    end
+  end
+
+  describe '#build_plane' do
+    it 'creates new plane in airport\'s dock' do
+      expect(airport.planes).to be_empty
+      airport.build_plane
```
Comment (ddosque) [inline spec/airport_spec.rb]: I'm pretty sure airports don't build planes. Check the brief.

File: spec/airport_spec.rb [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+      expect(airport.planes.count).to eq 1
+      expect(airport.planes[0]).to be_an_instance_of(Plane)
+    end
+
+    it 'takes an optional argument that specifies a custom plane' do
+      special_plane = double(:plane, :is_a? => true)
```
Comment (ddosque) [inline spec/airport_spec.rb]: custom planes are not mentioned in the brief, don't stray from it - you would be wasting your customer's time.

File: spec/airport_spec.rb [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+        expect { airport.build_plane(@non_plane) }.to raise_error "Error: The airport can only build planes."
+      end
+    end
+        
+  end
+
+  describe '#conditions' do
+    before(:example) do
+      @sunny_climate = double(:climate, :is_a? => true, :conditions => :sunny)
+      @airport = Airport.new(@sunny_climate)
```
Comment (ddosque) [inline spec/airport_spec.rb]: nice use of injection. If you weren't checking for type, you could have had a simpler double.

**Comment by adc17** [comment]
Thanks @ddosque . For the record, I wasn't trying to be fancy—these are just bad solutions to the brief that I wish I could have done more simply. e.g. airports building planes—I couldn't think of another way to generate a plane inside an airport when weather was stormy, and I needed one inside to check whether it was prevented from leaving.

**Pull request closed without merging**